### PR TITLE
Fix security shop regex

### LINKED ISSUE #12377: The identifier id is missing for a query of App\Entity\Customer\Customer
**Sylius version affected**: 1.8.9

**Description**  

When you're logout and you go to the following route : /account/dashboard
It throw an error 500 with the following message : The identifier id is missing for a query of App\Entity\Customer\Customer

**Steps to reproduce**  
- Start your web server
- Check than you are logout
- Go to the /account/dashboard route
- You must see the HTTP 500 Error
- 
**Possible Solution**  
There's an old post, I don't know if it can help : https://github.com/Sylius/Sylius/issues/8801

### PR DESCRIPTION
**Pull request by mmenozzi**
| Q               | A
| --------------- | -----
| Branch?         | 1.9
| Bug fix?        | yes
| New feature?    | no
| BC breaks?      | no
| Deprecations?   | no
| Related tickets | fixes #12377, #13144  
| License         | MIT

As documented [here](https://docs.sylius.com/en/1.10/cookbook/shop/disabling-localised-urls.html), Sylius should work properly even if there is no locale in route paths. But with the current `sylius.security.shop_regex` many access control paths don't match when there is no locale in the route path.

This change fix this. The regex now matches even if the string before `/account` or `/login` is empty.

### CODE REVIEW

**Comment by Loocos** [comment]
@mmenozzi I just tried your fix today but it seems to not work for my case #12377. Must be something else

**Comment by mmenozzi** [comment]
I never tested on 1.8 but it works on 1.9 and 1.10. What's the actual value of the regex in your case?

**Comment by Loocos** [comment]
It was this first : 
`sylius.security.shop_regex: "^/(?!%sylius_admin.path_name%|new-api|api/.*|api$|media/.*)[^/]++"`
and I've tried your solution after :
`sylius.security.shop_regex: "^(?:/(?!%sylius_admin.path_name%|api/.*|api$|media/.*)[^/]++)?"`

**Comment by mmenozzi** [comment]
The regex provided in this PR works, you can check it [here](https://regex101.com/r/JpGM5G/1).

So it must be something else in your case.

Do you have locale in your routes? What's the error you get?

**Comment by mmenozzi** [comment]
The parameter `sylius.security.shop_regex` is also defined in the Sylius-Standard template ([here](https://github.com/Sylius/Sylius-Standard/blob/1.10/config/packages/security.yaml#L3)) and wins over the changed definition done here.

So when this PR will be merged the change should be also ported to the Sylius-Standard template. Or, maybe better, we could simply remove those duplicate parameters definitions. @lchrusciel what do you think?

**Comment by pocky** [comment]
Hello,

@mmenozzi Thanks for your work, I can confirm this bug is also on Sylius 1.10
@lchrusciel Could you review this PR please?

**Comment by stefandoorn** [comment]
I can confirm homepage was not covered by the original regexp, so this seems to fix that.

**Comment by vvasiloi** [comment]
I was a bit skeptical when I saw the change, so I tested it (see **Unit Tests** tab): https://regex101.com/r/isSmzT/1.
It seems to be matching all routes, including admin and API, not just shop.
But then I started a Sylius instance with the change and opened shop, admin and API pages and checked the firewall and it was the right one every time...
Turns out it works because the shop firewall is the last in the configuration, so it will be matched only if any other firewall declared before it doesn't match.
After I moved it to first position, I could reproduce the issue and any page I accessed was behind shop firewall.
In other words: this RegEx doesn't work.
Also, we probably need some tests for it.

**Comment by mmenozzi** [comment]
> I was a bit skeptical when I saw the change, so I tested it (see **Unit Tests** tab): https://regex101.com/r/isSmzT/1. It seems to be matching all routes, including admin and API, not just shop. But then I started a Sylius instance with the change and opened shop, admin and API pages and checked the firewall and it was the right one every time... Turns out it works because the shop firewall is the last in the configuration, so it will be matched only if any other firewall declared before it doesn't match. After I moved it to the first position, I could reproduce the issue and every page I accessed was behind shop firewall. In other words: this RegEx doesn't work. Also, we probably need some tests for it.

Hi @vvasiloi, I think that if we don't want to rely on firewall position the only way to have a working shop firewall regex is to list all possible paths of the shop.

Even the current regex works only because it's the last in the firewall definitions (see your Unit Test [here](https://regex101.com/r/uhr1FP/1) but with the current, broken, regex).

So, as you can see the problem of relying on firewall position is there even without this PR.

The difference is that if this PR will be merged and the firewall position will be kept those stores without locale in the URL paths will work properly.

**Comment by vvasiloi** [comment]
@mmenozzi I don't think we need to list all the possible shop paths, it's probably enough to properly exclude the path prefixes that are clearly not shop, like /admin and /api. We just need a better RegEx. 😄

**Comment by mmenozzi** [comment]
That's what the regex in this PR is trying to do. But it's not easy to create such regex that works in all possible situations.

Btw I think that the regex provided in this PR is better than the old one given the current sort order of firewall definition. It avoids bugs when the store has no locale in URL paths.

So I think that this PR should be merged.

**Comment by pocky** [comment]
Hello,

Is it possible to update the current documentation with a note about this problem? Could be a link to this PR, another line with this regex or something else. Real problem will not be fixed but people who don't have locales in urls will find a solution.

**Comment by jakubtobiasz** [comment]
Hi @mmenozzi!
Thank you very much for pointing this problem. Instead of merging your PR directly into the core, we decided to update the doc page you mentioned in your PR's description. Thank you once again :).
